container-search/src/main/java/com/yahoo/prelude/statistics/StatisticsSearcher.java

```diff
@@ -180,14 +170,43 @@ private void qps(Metric.Context metricContext) {
         return context;
     }
 
+    private Metric.Context getDegradedMetricContext(String chainName, Coverage coverage) {
+        Map<DegradedReason, Metric.Context> reasons = degradedReasonContexts.get(chainName);
+        if (reasons == null) {
+            DegradedReason [] reasonEnums = {
+                    DegradedReason.MatchPhase, DegradedReason.AdaptiveTimeout,
+                    DegradedReason.Timeout, DegradedReason.NonIdealState
+            };
+            reasons = new HashMap<>(4);
+            for (DegradedReason reason : reasonEnums ) {
+                Map<String, String> dimensions = new HashMap<>();
+                dimensions.put("chain", chainName);
+                dimensions.put("reason", reason.toString());
+                Metric.Context context = this.metric.createContext(dimensions);
+                reasons.put(reason, context);
+            }
+            degradedReasonContexts.put(chainName, reasons);
+        }
+        if (coverage.isDegradedByMatchPhase()) {
```

Comment:
I suggest you move DegradedReason to Coverage and create an accessor for it. Then you can replace these 10 lines by a single lookup.

Reply:
That was my initial thought too. However as the reasons are not exclusive, and here we just select one of them, I concluded that it was better to have them here.
Ideally we should report all, but I think that will be more confusing.